spec/expression.dd

```diff
@@ -677,13 +670,24 @@ $(H3 $(LEGACY_LNAME2 class_comparisons, class-comparisons, Class comparisons))
     $(P For class objects, the relational
         operators compare the
         contents of the objects. Therefore, comparing against
-        $(CODE null) is invalid, as $(CODE null) has no contents.)
+        a $(CODE null) class reference is invalid, as $(CODE null) has no contents.)
 
+        $(SPEC_RUNNABLE_EXAMPLE_FAIL
         ---
-        class C;
+        class C {}
         C c;
-        if (c < null)  // error
-            ...
+        if (c < null) {}  // compile-time error
```

Comment:
Shouldn't this be put in a `void main() {}`? A top-level if-statement doesn't parse.

Reply:
Examples that don't contain `main` are wrapped in a `main` function (and have `stdio` imported). So C here is a nested class.

Reply:
Good to know